dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentTypeLoader.java

```diff
@@ -0,0 +1,303 @@
+package com.dotcms.rendering.velocity.services;
+
+
+
+import com.dotcms.contenttype.business.ContentTypeAPI;
+import com.dotcms.contenttype.model.field.CheckboxField;
+import com.dotcms.contenttype.model.field.DateField;
+import com.dotcms.contenttype.model.field.DateTimeField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.FileField;
+import com.dotcms.contenttype.model.field.ImageField;
+import com.dotcms.contenttype.model.field.RadioField;
+import com.dotcms.contenttype.model.field.SelectField;
+import com.dotcms.contenttype.model.field.TextAreaField;
+import com.dotcms.contenttype.model.field.TextField;
+import com.dotcms.contenttype.model.field.TimeField;
+import com.dotcms.contenttype.model.field.WysiwygField;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.File;
+import java.io.InputStream;
+import java.util.List;
+
+import org.apache.velocity.runtime.resource.ResourceManager;
+
+/**
+ * @author will
+ */
+public class ContentTypeLoader implements DotLoader {
+
+    private long fakeIdentifier = Long.MAX_VALUE;
+    private long fakeInode = Long.MAX_VALUE;
+    private String fakeTitle = "Content Title";
+
+
+
+    public void invalidate(ContentType contentType, PageMode mode) {
+        removeContentTypeFile(contentType);
+    }
+
+
+
+    public InputStream buildVelocity(ContentType type, String filePath, PageMode mode ) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 26 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "mode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This method has 188 lines, which is greater than the 75 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS138)